daemons/controld/controld_te_actions.c

```diff
@@ -264,7 +264,8 @@ controld_record_action_event(pcmk__graph_action_t *action,
 
     int target_rc = get_target_rc(action);
 
-    action_rsc = find_xml_node(action->xml, PCMK_XE_PRIMITIVE, TRUE);
+    action_rsc = pcmk__xe_first_child(action->xml, PCMK_XE_PRIMITIVE, NULL,
+                                      NULL);
     if (action_rsc == NULL) {
```

Comment:
A warning wouldn't hurt, but it doesn't really matter. This would have to be a malformed graph.

Reply:
If it's a malformed graph, that sounds more like a cause for a `CRM_CHECK()` due to likely bug. I'll leave it alone for the moment.

Reply:
Sounds good